### ISSUE TITLE
Fermentrack frequently fails to load properly, warns that "Another isntance of BrewPi is already running ... This instance will exit"

### ISSUE THREAD

**Issue by mwanderman** [thorrak/fermentrack#717]
Fermentrack frequently fails to load properly. I put together a new Raspberry Pi 3B the other day and setup Fermentrack. More often than not, it loads with the message `Cannot receiveLCD text fromController/Script`.

Here is the controller's stderr log from the latest attempt:
```
 Jun 01 2023 17:03:41   Another instance of BrewPi is already running, which will conflict with this instance. This instance will exit
 Jun 01 2023 17:04:21   Refreshing dbConfig
 Jun 01 2023 17:04:21   Another instance of BrewPi is already running, which will conflict with this instance. This instance will exit
 Jun 01 2023 17:04:31   Refreshing dbConfig
 Jun 01 2023 17:04:31   Another instance of BrewPi is already running, which will conflict with this instance. This instance will exit
 Jun 01 2023 17:04:42   Refreshing dbConfig
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/base/base.py", line 220, in ensure_connection
    self.connect()
  File "/usr/local/lib/python3.9/site-packages/sentry_sdk/integrations/django/__init__.py", line 579, in connect
    return real_connect(self)
  File "/usr/local/lib/python3.9/site-packages/django/utils/asyncio.py", line 26, in inner
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/base/base.py", line 197, in connect
    self.connection = self.get_new_connection(conn_params)
  File "/usr/local/lib/python3.9/site-packages/django/utils/asyncio.py", line 26, in inner
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/postgresql/base.py", line 185, in get_new_connection
    connection = Database.connect(**conn_params)
  File "/usr/local/lib/python3.9/site-packages/psycopg2/__init__.py", line 127, in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
psycopg2.OperationalError: FATAL:  the database system is shutting down


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/local/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/app/brewpi-script/brewpi.py", line 159, in <module>
    dbConfig = refresh_dbConfig()
  File "/app/brewpi-script/brewpi.py", line 109, in refresh_dbConfig
    return models.BrewPiDevice.objects.get(id=device_id)
  File "/usr/local/lib/python3.9/site-packages/django/db/models/manager.py", line 82, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/django/db/models/query.py", line 411, in get
    num = len(clone)
  File "/usr/local/lib/python3.9/site-packages/django/db/models/query.py", line 258, in __len__
    self._fetch_all()
  File "/usr/local/lib/python3.9/site-packages/django/db/models/query.py", line 1261, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
  File "/usr/local/lib/python3.9/site-packages/django/db/models/query.py", line 57, in __iter__
    results = compiler.execute_sql(chunked_fetch=self.chunked_fetch, chunk_size=self.chunk_size)
  File "/usr/local/lib/python3.9/site-packages/django/db/models/sql/compiler.py", line 1152, in execute_sql
    cursor = self.connection.cursor()
  File "/usr/local/lib/python3.9/site-packages/django/utils/asyncio.py", line 26, in inner
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/base/base.py", line 260, in cursor
    return self._cursor()
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/base/base.py", line 236, in _cursor
    self.ensure_connection()
  File "/usr/local/lib/python3.9/site-packages/django/utils/asyncio.py", line 26, in inner
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/base/base.py", line 220, in ensure_connection
    self.connect()
  File "/usr/local/lib/python3.9/site-packages/django/db/utils.py", line 90, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/base/base.py", line 220, in ensure_connection
    self.connect()
  File "/usr/local/lib/python3.9/site-packages/sentry_sdk/integrations/django/__init__.py", line 579, in connect
    return real_connect(self)
  File "/usr/local/lib/python3.9/site-packages/django/utils/asyncio.py", line 26, in inner
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/base/base.py", line 197, in connect
    self.connection = self.get_new_connection(conn_params)
  File "/usr/local/lib/python3.9/site-packages/django/utils/asyncio.py", line 26, in inner
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/postgresql/base.py", line 185, in get_new_connection
    connection = Database.connect(**conn_params)
  File "/usr/local/lib/python3.9/site-packages/psycopg2/__init__.py", line 127, in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
django.db.utils.OperationalError: FATAL:  the database system is shutting down

Sentry is attempting to send 2 pending error messages
Waiting up to 2 seconds
Press Ctrl-C to quit
 Jun 01 2023 17:06:10   Refreshing dbConfig
 Jun 01 2023 17:06:10   Another instance of BrewPi is already running, which will conflict with this instance. This instance will exit
 Jun 01 2023 17:06:19   Refreshing dbConfig
 Jun 01 2023 17:06:19   Another instance of BrewPi is already running, which will conflict with this instance. This instance will exit
 Jun 01 2023 17:06:29   Refreshing dbConfig
 Jun 01 2023 17:06:29   Another instance of BrewPi is already running, which will conflict with this instance. This instance will exit
 Jun 01 2023 17:06:38   Refreshing dbConfig
 Jun 01 2023 17:06:38   Another instance of BrewPi is already running, which will conflict with this instance. This instance will exit
 Jun 01 2023 17:06:48   Refreshing dbConfig
 Jun 01 2023 17:06:49   Another instance of BrewPi is already running, which will conflict with this instance. This instance will exit
 Jun 01 2023 17:06:58   Refreshing dbConfig
 Jun 01 2023 17:06:59   Another instance of BrewPi is already running, which will conflict with this instance. This instance will exit
 Jun 01 2023 17:07:08   Refreshing dbConfig
 Jun 01 2023 17:07:09   Another instance of BrewPi is already running, which will conflict with this instance. This instance will exit
 Jun 01 2023 17:07:19   Refreshing dbConfig
 Jun 01 2023 17:07:19   Another instance of BrewPi is already running, which will conflict with this instance. This instance will exit
 Jun 01 2023 17:07:29   Refreshing dbConfig
 Jun 01 2023 17:07:29   Another instance of BrewPi is already running, which will conflict with this instance. This instance will exit
 Jun 01 2023 17:07:39   Refreshing dbConfig
 Jun 01 2023 17:07:39   Another instance of BrewPi is already running, which will conflict with this instance. This instance will exit
 Jun 01 2023 17:07:49   Refreshing dbConfig
 Jun 01 2023 17:07:49   Another instance of BrewPi is already running, which will conflict with this instance. This instance will exit
```

As far as I know I don't have any other instances of BrewPi or Fermentrack set up. The Pi is dedicated as a fermentation chamber controller and has no other use.

**Comment by thorrak** [comment]
I've seen that before, but typically as a race condition when Fermentrack is starting up. Once the database server fully comes up it stabilizes, and everything else starts to work as well. This process takes _much_ longer on the first boot, but should be fine after that.

Am I correct to assume that Fermentrack never stabilizes, and you continue to see errors well after it should have finished booting?

**Comment by mwanderman** [comment]
Any sense of how long "_much_ longer" is? I usually give it a couple minutes before restarting. Last night I posted the issue and went to bed. When I woke in the morning it seemed to be working fine.

**Comment by thorrak** [comment]
10-20 minutes for slower Pi’s/SD cards, but only the first boot after you install Fermentrack. After that, maybe 1-2 minutes per boot if things are particularly slow. 

Once things have finished booting and stabilized, they should stay stabilized, however. Have you been shutting the Pi down?

**Comment by mwanderman** [comment]
Thanks. I have shut down or restarted the Pi at times. I don't see anything in the Fermentrack interface for a clean shutdown. What is the correct way to do so?

**Comment by HuggableShark** [comment]
Been receiving the same error here as of today as well. Tried docker restarts, full pi and controller power flushes, just power flushing the esp8266, and standard software reboots of the pi. No changes. Controller is connected via USB cable for a serial connection in my instance. Was working fine before I had to reboot the pi earlier today and  can't seem to get it to recover since. Been using the same containers pulled from latest for the past 11 days.

` Jun 15 2023 21:55:26   Another instance of BrewPi is already running, which will conflict with this instance. This instance will exit
 Jun 15 2023 21:55:36   Refreshing dbConfig
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/base/base.py", line 220, in ensure_connection
    self.connect()
  File "/usr/local/lib/python3.9/site-packages/sentry_sdk/integrations/django/__init__.py", line 579, in connect
    return real_connect(self)
  File "/usr/local/lib/python3.9/site-packages/django/utils/asyncio.py", line 26, in inner
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/base/base.py", line 197, in connect
    self.connection = self.get_new_connection(conn_params)
  File "/usr/local/lib/python3.9/site-packages/django/utils/asyncio.py", line 26, in inner
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/postgresql/base.py", line 185, in get_new_connection
    connection = Database.connect(**conn_params)
  File "/usr/local/lib/python3.9/site-packages/psycopg2/__init__.py", line 127, in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
psycopg2.OperationalError: FATAL:  the database system is shutting down


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/local/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/app/brewpi-script/brewpi.py", line 159, in <module>
    dbConfig = refresh_dbConfig()
  File "/app/brewpi-script/brewpi.py", line 109, in refresh_dbConfig
    return models.BrewPiDevice.objects.get(id=device_id)
  File "/usr/local/lib/python3.9/site-packages/django/db/models/manager.py", line 82, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/django/db/models/query.py", line 411, in get
    num = len(clone)
  File "/usr/local/lib/python3.9/site-packages/django/db/models/query.py", line 258, in __len__
    self._fetch_all()
  File "/usr/local/lib/python3.9/site-packages/django/db/models/query.py", line 1261, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
  File "/usr/local/lib/python3.9/site-packages/django/db/models/query.py", line 57, in __iter__
    results = compiler.execute_sql(chunked_fetch=self.chunked_fetch, chunk_size=self.chunk_size)
  File "/usr/local/lib/python3.9/site-packages/django/db/models/sql/compiler.py", line 1152, in execute_sql
    cursor = self.connection.cursor()
  File "/usr/local/lib/python3.9/site-packages/django/utils/asyncio.py", line 26, in inner
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/base/base.py", line 260, in cursor
    return self._cursor()
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/base/base.py", line 236, in _cursor
    self.ensure_connection()
  File "/usr/local/lib/python3.9/site-packages/django/utils/asyncio.py", line 26, in inner
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/base/base.py", line 220, in ensure_connection
    self.connect()
  File "/usr/local/lib/python3.9/site-packages/django/db/utils.py", line 90, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/base/base.py", line 220, in ensure_connection
    self.connect()
  File "/usr/local/lib/python3.9/site-packages/sentry_sdk/integrations/django/__init__.py", line 579, in connect
    return real_connect(self)
  File "/usr/local/lib/python3.9/site-packages/django/utils/asyncio.py", line 26, in inner
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/base/base.py", line 197, in connect
    self.connection = self.get_new_connection(conn_params)
  File "/usr/local/lib/python3.9/site-packages/django/utils/asyncio.py", line 26, in inner
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/postgresql/base.py", line 185, in get_new_connection
    connection = Database.connect(**conn_params)
  File "/usr/local/lib/python3.9/site-packages/psycopg2/__init__.py", line 127, in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
django.db.utils.OperationalError: FATAL:  the database system is shutting down

Sentry is attempting to send 2 pending error messages
Waiting up to 2 seconds
Press Ctrl-C to quit
 Jun 15 2023 21:55:46   Refreshing dbConfig
 Jun 15 2023 21:55:46   Another instance of BrewPi is already running, which will conflict with this instance. This instance will exit
 Jun 15 2023 21:55:56   Refreshing dbConfig
 Jun 15 2023 21:55:56   Another instance of BrewPi is already running, which will conflict with this instance. This instance will exit
 Jun 15 2023 21:56:06   Refreshing dbConfig
 Jun 15 2023 21:56:06   Another instance of BrewPi is already running, which will conflict with this instance. This instance will exit
 Jun 15 2023 21:56:16   Refreshing dbConfig
 Jun 15 2023 21:56:16   Another instance of BrewPi is already running, which will conflict with this instance. This instance will exit`

**Comment by HuggableShark** [comment]
Ran a `docker-compose down` followed by a `docker-compose up -d` from the root directory and that seemed to get communication back up and running. No data loss as well. Not sure if it reproducible yet or not, but it worked in my case.

**Comment by thorrak** [comment]
Yeah, I have no idea. I’m seeing errors reported in Sentry that look related, but they all seem to relate to Tilt support of all things. 

I actually have rebuilding Tilt support into its own Docker image on my agenda - I’ll take a look at that later and we can see if either that - or just rebuilding the images - potentially helps here. My guess is that it won’t, but it will remove one potential error from the mix at least.

**Comment by thorrak** [comment]
I'm back home after traveling last week, and am building some new Fermentrack instances to try to test this. More info coming soon, I hope.

**Comment by thorrak** [comment]
After spinning up the new Fermentrack instances, I realized that there's actually two very different versions of BrewPi script in use at the moment. @HuggableShark and @mwanderman - when you click on the "gear" icon and then click "Update from GitHub" - What do you see listed next to "Active Branch"? `dev`, `docker-dev`, `master`, or something else?

**Comment by mwanderman** [comment]
I'm on `master`

Local Commit Info:
Active Branch:	master
Commit Date:	April 6, 2022, 9:54 p.m.
Commit Message:	
Merge branch 'dev' of github.com:thorrak/fermentrack
Remote Commit Info:
Commit Date:	April 6, 2022, 9:54 p.m.
Commit Message:	
Merge branch 'dev' of github.com:thorrak/fermentrack

**Comment by thorrak** [comment]
OK, good -- that gives me some hope that the error relates to the older circus/brewpi-script configuration and might be solved by the new code. If you're interested in trying out the new branch to see if it fixes things, here's how you can do that:

## Create a Backup

Before you do anything, back up your Fermentrack data:

1. Log into Fermentrack and click the "gear" in the upper right corner
2. At the bottom of the page, click "List/Create Backups"
3. Click "Create New Backup"
4. Click "Add Backup"
5. Click "Download Backup" and make sure a file downloads, which should end in ".tar.xz"

## Force-switch to the "testing" docker image

**NOTE** - Unfortunately, progress here is a bit of a one-way street. Once you've executed the commands below, you will be on the `testing` image which contains a number of database changes which means that reverting to the previous `master` branch is very much not advisable. Later, once this branch is released, you will also need to manually repeat the steps below, substituting `latest` for `testing`. You'll know when this happens, as I will close this issue out once the code is merged. 

1. Log into your Raspberry Pi via SSH
2. Change to the fermentrack-tools directory (typically `cd fermentrack-tools`)
3. Run `./install.sh -n -i testing`
4. *If prompted* type your root password (likely won't be necessary)
5. Wait for the installation to complete (until you see the "Done Installing Fermentrack" message)

Given the one way nature of the above, if you'd rather wait until this code gets merged into master, that's fine as well! It may be a few weeks, however, as I need to get scripts together to allow testing

**Comment by HuggableShark** [comment]
@thorrak Sorry for the late response, I see you've been busy with a bunch of development and issue responses over the last few days so thanks for the good work!

Current version I'm running shows as follows:
```
Active Branch: dev
Commit Date Oct. 17, 2022
```

I just went through the upgrade from local install to docker install about 1 month ago. I copied over my backup from that time, which may be why it looks like my dev install is so old.

**Comment by thorrak** [comment]
> @thorrak Sorry for the late response, I see you've been busy with a bunch of development and issue responses over the last few days so thanks for the good work!
> 
> Current version I'm running shows as follows:
> 
> ```
> Active Branch: dev
> Commit Date Oct. 17, 2022
> ```
> 
> I just went through the upgrade from local install to docker install about 1 month ago. I copied over my backup from that time, which may be why it looks like my dev install is so old.

It's not -- that's actually when the "dev" (and "master"!) images were last built, unfortunately! I've apparently been working on this update for awhile.

The instructions I posted above should bring you to the new "testing" branch, but the same warning applies. Make a backup first if you want to try it out, as there it won't be easy going back if you choose to. That said, I'm probably going to release this code in the not-too-distant future, assuming I can make a decision as to whether or not to try to include the new tilt monitor in this release.

**Comment by HuggableShark** [comment]
I can assist if you want me to test anything. I can always pull and spin up a second stable instance when I begin my brew this coming weekend if there are issues. Currently it's functioning fine otherwise and I just down and up the containers if it acts up.

**Comment by thorrak** [comment]
If you don't mind following the instructions [here](https://github.com/thorrak/fermentrack/issues/717#issuecomment-1596173019) then and letting me know if this solves the issue, I would appreciate it!

**Comment by HuggableShark** [comment]
Completed. Error hasn't shown since I re-created the container last, but no error after the update either. Anything you'd like me to try to push it to see if an error generates?

**Comment by thorrak** [comment]
That's good news, I think! Hopefully it stays stable. If you notice anything strange, let me know.

**Comment by HuggableShark** [comment]
Tried to re-apply my https adjustment and re-built the containers today. After doing so all containers started normally, but I think Django is hung up on the new tilt_monitor during it startup.

```
0 static files copied to '/app/collected_static', 272 unmodified.
Operations to perform:
  Apply all migrations: admin, app, auth, backups, contenttypes, database, external_push, firmware_flash, gravity, sessions
Running migrations:
  No migrations to apply.
2023-06-19 13:35:18,747 INFO Included extra file "/etc/supervisor/conf.d/supervisord.conf" during parsing
Unlinking stale socket /var/run/supervisor.sock
2023-06-19 13:35:19,063 INFO RPC interface 'supervisor' initialized
2023-06-19 13:35:19,064 CRIT Server 'unix_http_server' running without any HTTP authentication checking
2023-06-19 13:35:19,065 INFO supervisord started with pid 26
2023-06-19 13:35:20,074 INFO spawned: 'brewpi_script_monitor' with pid 27
2023-06-19 13:35:20,085 INFO spawned: 'gunicorn' with pid 28
2023-06-19 13:35:20,095 INFO spawned: 'huey' with pid 29
2023-06-19 13:35:20,106 INFO spawned: 'tilt_monitor' with pid 30
2023-06-19 13:35:21,111 INFO success: brewpi_script_monitor entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2023-06-19 13:35:21,112 INFO success: gunicorn entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2023-06-19 13:35:21,112 INFO success: huey entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2023-06-19 13:35:21,113 INFO success: tilt_monitor entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2023-06-19 13:35:37,045 INFO exited: tilt_monitor (exit status 1; not expected)
2023-06-19 13:35:37,052 INFO spawned: 'tilt_monitor' with pid 74
2023-06-19 13:35:38,127 INFO success: tilt_monitor entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2023-06-19 13:35:43,738 INFO exited: tilt_monitor (exit status 1; not expected)
2023-06-19 13:35:44,747 INFO spawned: 'tilt_monitor' with pid 81
2023-06-19 13:35:45,751 INFO success: tilt_monitor entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2023-06-19 13:35:52,092 INFO exited: tilt_monitor (exit status 1; not expected)
2023-06-19 13:35:53,099 INFO spawned: 'tilt_monitor' with pid 89
2023-06-19 13:35:54,102 INFO success: tilt_monitor entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2023-06-19 13:35:59,462 INFO exited: tilt_monitor (exit status 1; not expected)
2023-06-19 13:36:00,471 INFO spawned: 'tilt_monitor' with pid 96
2023-06-19 13:36:01,476 INFO success: tilt_monitor entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2023-06-19 13:36:07,244 INFO exited: tilt_monitor (exit status 1; not expected)
2023-06-19 13:36:08,254 INFO spawned: 'tilt_monitor' with pid 103
2023-06-19 13:36:09,259 INFO success: tilt_monitor entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2023-06-19 13:36:15,703 INFO exited: tilt_monitor (exit status 1; not expected)
2023-06-19 13:36:16,713 INFO spawned: 'tilt_monitor' with pid 110
2023-06-19 13:36:17,718 INFO success: tilt_monitor entered RUNNING state, process has stayed up for > than 1 seconds 
```

nginx is working fine, just has nothing to direct  the webtraffic to which led me to the assumption about Django
![image](https://github.com/thorrak/fermentrack/assets/30677760/1967895d-d3c8-4bb0-a4cb-a42aade037cb)

We can open a different issue for this if you would prefer as well. I'll try re-building after giving it 15 minutes to hopefully finish setting up.

**Comment by thorrak** [comment]
In your `nginx.conf` what port do you have listed for the HTTPS redirect? 8123 or 5000?

I think your PR had it as 8123 which was the correct port with Circus -- for this branch, it now needs to be 5000.

**Comment by HuggableShark** [comment]
You called it, I had 8123. Changed to 5000 for both http and https redirects, rebuilt and it loaded like a charm. I can make the issue if you'd like for documentation, but that was def a resolution. I'll work on a pull request to the fermentrack-tools repo with a script to smooth out  adding those certs for users later today.

Running on a Rpi 3B+, no extraneous virtualization.

**Comment by thorrak** [comment]
Glad to hear that worked! No worries about raising an issue -- I merged your PR into the dev branch, and [already updated ](https://github.com/thorrak/fermentrack/commit/dc18f37ed994220df1ff55843b600a0a2e914a85)it to match the new port. If you have any changes to fermentrack-tools, however, those are definitely welcome -- I'll end up syncing any releases between the two repos.

On the tilt_monitor thing - it's good to know that you are seeing the same issues on a Pi that I was seeing on my VM. I assumed that it related to the fact that I was running in a VM with no bluetooth adapter, but it sounds like that isn't the case. Thankfully, I'm about 60% of the way done with the new, standalone tilt monitor ("TiltBridge Junior"?) so hopefully I can get that included as part of the release as well.

**Comment by HuggableShark** [comment]
Sounds great, I'll push that after I get some time to work on it this afternoon later today then.

For the tilt-monitor, I have  bluetooth power disabled on my RPI, so it COULD possibly be related. If needed I could try to reactivate it and see how the django container's logs alter.

**Comment by thorrak** [comment]
Hah - that makes me feel a lot better, actually. Don’t worry about reenabling it - I’m going to need to run tests on my side regardless.

**Comment by mwanderman** [comment]
I'm busy for the next few weeks plus my ferm chamber is busy lagering, so I'll wait for the master update.

**Comment by thorrak** [comment]
Everything has been pushed to master and can be used by following the instructions in #727 . I’m expecting that this should resolve the problems in this thread — but even if it doesn’t, given the fact that the updated code is substantially rewritten, I would request that a new issue be opened to track against. 

@HuggableShark - Your HTTPS support will get nuked by these instructions. I will look to add better support to Fermentrack Tools for this in the future (which I will track against [in that repo](https://github.com/thorrak/fermentrack-tools/issues/54) ) but for now I recommend copying your certificate files and modified NGINX config out of the compose folder, running the install.sh script, copying the files back, and then running the docker-upgrade script manually.
